api/Controllers/ImagesController.cs

```diff
@@ -0,0 +1,197 @@
+        public async Task<IActionResult> AddHotelImage([FromRoute] int hotelId)
+        {
+            if (!CheckIfHotelExists(hotelId))
+            {
+                return BadRequest("Such hotel does not exist");
+            }
+
+            string fileName = null;
+            try
+            {
+                var file = Request.Form.Files[0];
+                fileName = await SaveFileAsync(file);
+            }
+            catch
+            {
+                return BadRequest("Something is wrong with file, probably file is empty");
+            }
+            ImageEntity image = new()
+            {
+                Path = fileName,
+                HotelId = hotelId,
+                IsOuterLink = false,
+            };
+            await _hotelsDb.Images.AddAsync(image);
+            await _hotelsDb.SaveChangesAsync();
+            return CreatedAtAction
+            (
+                nameof(GetImage), 
+                new { hotelId = image.HotelId, imageId = image.Id }, 
+                image.Id
+            );
+        }
+
+        [Route("{hotelId}/rooms/{roomId}/images")]
+        [HttpPost]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> AddRoomImage([FromRoute] int hotelId, [FromRoute] int roomId)
+        {
+            if (!CheckIfHotelExists(hotelId))
+            {
+                return BadRequest("Such hotel does not exist");
+            }
+            if (!CheckIfRoomExists(roomId))
+            {
+                return BadRequest("Such room does not exist");
+            }
+            var file = Request.Form.Files[0];
+            if (file.Length == 0)
+            {
+                return BadRequest("Something is wrong with file, probably file is empty");
+            }
+
+            var fileName = await SaveFileAsync(file);
+            ImageEntity image = new()
+            {
+                Path = fileName,
+                HotelId = hotelId,
+                IsOuterLink = false,
+                RoomId = roomId,
+            };
+            await _hotelsDb.Images.AddAsync(image);
+            await _hotelsDb.SaveChangesAsync();
+
+            return CreatedAtAction
+            (
+                nameof(GetImage), 
+                new { hotelId = image.HotelId, roomId = image.RoomId, imageId = image.Id }, 
+                image.Id
+            );
+        }
+
+        [Route("{hotelId}/images/{imageId}")]
+        [Route("{hotelId}/rooms/{roomId}/images/{imageId}")]
+        [HttpDelete]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> DeleteImage([FromRoute] int imageId)
+        {
+            ImageEntity image = await GetImageEntityAsync(imageId);
+            if (image == null)
+            {
+                return NotFound("Such image does not exist");
+            }
+            string imageFullPath = Path.Combine(_imagesFolder, image.Path);
+            if (System.IO.File.Exists(imageFullPath))
```

Comment:
`using System.IO`?

Reply:
There is another namespace in the class that also has File.Exists method